app/llm/performance_enhanced.py

```
@@ -0,0 +1,142 @@
+        try:
+            yield
+        except Exception as e:
+            logger.error(f"Operation '{operation_name}' failed: {e}")
+            with self._lock:
+                self._metrics.error_count += 1
+            raise
+        finally:
+            execution_time = time.perf_counter() - start_time
+            self._record_metrics(operation_name, execution_time)
+
+    def _record_metrics(self, operation_name: str, execution_time: float):
```

Ruff (0.8.2)

Missing return type annotation for private function `_record_metrics`

Add return type annotation: `None`

(ANN202)